src/core/reference/include/openvino/reference/roi_align.hpp

```diff
@@ -15,7 +15,189 @@ namespace ov {
 namespace reference {
 using ROIPoolingMode = op::v3::ROIAlign::PoolingMode;
 using AlignedMode = op::v9::ROIAlign::AlignedMode;
+
+namespace roi_policy {
+
+// Base class for representing sampling space.
+// Sampling space is defined as aligned bounding box.
+template <typename T>
+class SamplingSpaceBase {
+public:
+    T get_start_x() const {
+        return start_x;
+    }
+    T get_start_y() const {
+        return start_y;
+    }
+    T get_size_x() const {
+        return size_x;
+    }
+    T get_size_y() const {
+        return size_y;
+    }
+
+protected:
+    SamplingSpaceBase(T start_x_, T start_y_, T size_x_, T size_y_)
+        : start_x(start_x_),
+          start_y(start_y_),
+          size_x(size_x_),
+          size_y(size_y_) {}
+
+    T start_x;
+    T start_y;
+    T size_x;
+    T size_y;
+};
+
+template <typename T>
+struct Point {
+    T x;
+    T y;
+};
+
+template <typename T>
+class ROIAlignOpDefPolicy {
```

Comment:
Can you explain if this class do any changes/improvements to for ROI align? 
Or it is just to have like common interface for old and new versions.

If there is some improvement maybe some test should be added to avoid accidental changes during re-factoring e.g. for bin-size reduction.

Just curios did you check how these changes impact on current binary-size openvino lib (decrease/increase)?

Reply:
> Can you explain if this class do any changes/improvements to for ROI align?
> Or it is just to have like common interface for old and new versions.
> 
> If there is some improvement maybe some test should be added to avoid accidental changes during re-factoring e.g. for bin-size reduction.

Honestly I don't understand your question. If you ask about policy design in general - what are prons and cons, please refer to book "Modern C++ Design" written by by Andrei Alexandrescu - I very good book on the topic.

> Just curios did you check how these changes impact on current binary-size openvino lib (decrease/increase)?

I haven't made any checks, but I am afraid each time you add a new code - some lib/exe binary size will increase. And yes, this add a new code and yet another template instantiation of roi_align template, so some binary size(s) will increase.